data/core/macros/utils.cfg

```diff
@@ -534,50 +534,29 @@
             [/and]
         [/filter_condition]
 
-        [foreach]
-            array=unit.attack
-            [do]
-                [if]
-                    #This is to mute a warning message about retrieving a member of non-existant wml container.
-                    [variable]
-                        name=this_item.specials.length
-                        greater_than=0
-                    [/variable]
-
-                    [variable]
-                        name=this_item.specials.chance_to_hit.length
-                        greater_than=0
-                    [/variable]
-
-                    [then]
-                        [set_variables]
-                            name=this_item.specials.original_chance_to_hit
-                            to_variable=this_item.specials.chance_to_hit
-                        [/set_variables]
-
-                        {CLEAR_VARIABLE this_item.specials.chance_to_hit}
-                    [/then]
-                [/if]
-
-                [set_variables]
-                    name=this_item.specials.chance_to_hit
-
-                    [value]
-                        id=forced_cth
-                        value={CTH_NUMBER}
-                        cumulative=no
-                        [filter_opponent]
-                            {SECOND_FILTER}
-                        [/filter_opponent]
-                    [/value]
-                [/set_variables]
-            [/do]
-        [/foreach]
-
-        [unstore_unit]
-            variable=unit
-            find_vacant=no
-        [/unstore_unit]
+        [object]
+		id=forced_cth
+		[filter]
+			{FILTER}
+		[/filter]
```

Comment:
[object][filter] gives the object only to the _first_ unit matching the filter, so your change breaks the macro in the case that {FILTER} matches more than one unit.